include/oneapi/dpl/pstl/parallel_backend.h

```diff
@@ -15,6 +15,20 @@
 #ifndef _ONEDPL_PARALLEL_BACKEND_H
 #define _ONEDPL_PARALLEL_BACKEND_H
 #include "onedpl_config.h"
+
+// Select a parallel backend
+#if ONEDPL_USE_TBB_BACKEND || (!defined(ONEDPL_USE_TBB_BACKEND) && !ONEDPL_USE_OPENMP_BACKEND && _ONEDPL_TBB_AVAILABLE)
+#    define _ONEDPL_PAR_BACKEND_TBB 1
+#endif
+
+#if ONEDPL_USE_OPENMP_BACKEND || (!defined(ONEDPL_USE_OPENMP_BACKEND) && _ONEDPL_OPENMP_AVAILABLE)
+#    define _ONEDPL_PAR_BACKEND_OPENMP 1
+#endif
+
+#if !_ONEDPL_PAR_BACKEND_TBB && !_ONEDPL_PAR_BACKEND_OPENMP
+#    define _ONEDPL_PAR_BACKEND_SERIAL 1
+#endif
```

Comment:
Now it's easy to move the `#if ... #elif ... #else` logic from below to these conditions, as well as add proper #includes, like this:
```
#if <conditions to use TBB backend>
#define _ONEDPL_PAR_BACKEND_TBB 1
#include "parallel_backend_tbb.h"
#elif <conditions to use OpenMP backend>
#define _ONEDPL_PAR_BACKEND_OPENMP 1
#include "parallel_backend_omp.h"
#else
#define _ONEDPL_PAR_BACKEND_SERIAL 1
# include "parallel_backend_serial.h"
#endif
```
And after that, the code below can be reorganized like this:
```
namespace oneapi { namespace dpl {
namespace __par_backend =
#if _ONEDPL_PAR_BACKEND_TBB
    __tbb_backend;
#elif ... // etc.
#endif
}} //namespaces
```

Reply:
Fixed.